tests/api/test_checkout.py

```diff
@@ -1776,3 +1778,205 @@ def test_is_fully_paid_no_payment(checkout_with_item):
+        update_checkout_meta, checkout_meta_update_variables, permissions=[]
+    )
+    content = get_graphql_content(response)
+    errors = content["data"]["checkoutUpdateMetadata"]["errors"]
+    assert len(errors) == 0
+    checkout.refresh_from_db()
+    assert checkout.get_meta(namespace="test", client="client1") == {"foo": "bar"}
+
+
+def test_user_with_permission_can_update_private_meta(
+    permission_manage_checkouts,
+    staff_api_client,
+    staff_user,
+    update_checkout_private_meta,
+    checkout_private_meta_update_variables,
+    checkout,
+):
+    staff_user.user_permissions.add(permission_manage_checkouts)
+    assert staff_user.has_perm(CheckoutPermissions.MANAGE_CHECKOUTS)
+    response = staff_api_client.post_graphql(
+        update_checkout_private_meta, checkout_private_meta_update_variables
+    )
+    content = get_graphql_content(response)
+    errors = content["data"]["checkoutUpdatePrivateMetadata"]["errors"]
+    assert len(errors) == 0
+    checkout.refresh_from_db()
+    assert checkout.get_private_meta(namespace="test", client="client1") == {"foo": "bar"}
+
+
+@pytest.fixture
+def clear_checkout_meta():
+    return """
+        mutation checkoutClearMetadata($token: UUID!, $input: MetaPath!) {
+            checkoutClearMetadata(token: $token, input: $input) {
+                errors {
+                    message
+                }
+            }
+        }
+    """
+
+
+@pytest.fixture
+def clear_checkout_private_meta():
+    return """
+        mutation checkoutClearPrivateMetadata($id: ID!, $input: MetaPath!) {
+            checkoutClearPrivateMetadata(id: $id, input: $input) {
+                errors {
+                    message
+                }
+            }
+        }
+    """
+
+
+@pytest.fixture
+def clear_meta_variables(checkout):
+    return {
+        "token": checkout.token,
+        "input": {"namespace": "test", "clientName": "client1", "key": "foo"},
+    }
+
+
+@pytest.fixture
+def clear_meta_private_variables(checkout):
+    return {
+        "id": checkout.pk,
+        "input": {
+            "namespace": "testPrivate",
+            "clientName": "clientPrivate1",
+            "key": "foo",
+        },
+    }
+
+
+@pytest.fixture
+def checkout_with_meta(checkout):
+    checkout.store_meta(namespace="test", client="client1", item={"foo": "bar"})
+    checkout.store_private_meta(
+        namespace="testPrivate", client="clientPrivate1", item={"foo": "bar"}
+    )
+    checkout.save()
+    return checkout
+
+
+def test_user_without_permission_cannot_clear_private_meta(
+    staff_user, staff_api_client, clear_checkout_private_meta, clear_meta_private_variables
+):
+    assert not staff_user.has_perm(CheckoutPermissions.MANAGE_CHECKOUTS)
+    response = staff_api_client.post_graphql(
+        clear_checkout_private_meta, clear_meta_private_variables
+    )
+    assert_no_permission(response)
+
+
+def test_user_with_permission_can_clear_meta(
```

Comment:
`test_staff_user_with_permission_can_clear_meta` would be more meaningfull

Reply:
I copied this from `test_order`, which are named the same, want me to rename that test to `test_staff*` also, or leave it for now and just rename this one?